plugins/soundsourcem4a/soundsourcem4a.cpp

```diff
@@ -382,91 +382,83 @@ void SoundSourceM4A::restartDecoding(MP4SampleId sampleBlockId) {
     m_sampleBuffer.reset();
 }
 
-SINT SoundSourceM4A::seekSampleFrame(SINT frameIndex) {
-    DEBUG_ASSERT(isValidFrameIndex(m_curFrameIndex));
-
-    DEBUG_ASSERT(frameIndex >= getMinFrameIndex());
-    if (frameIndex == m_curFrameIndex) {
-        return m_curFrameIndex;
-    }
-    if (frameIndex >= getMaxFrameIndex()) {
-        // EOF
-        m_curFrameIndex = getMaxFrameIndex();
-        return m_curFrameIndex;
-    }
-
-    // NOTE(uklotzde): Resetting the decoder near to the beginning
-    // of the stream when seeking backwards produces invalid sample
-    // values! As a consequence the seeking test fails.
-    if ((m_curSampleBlockId != MP4_INVALID_SAMPLE_ID) &&
-            (frameIndex < m_curFrameIndex) &&
-            (frameIndex <= (getMinFrameIndex() + kNumberOfPrefetchFrames))) {
-        // Workaround: Reset the decoder when seeking near to the beginning
-        // of the stream while decoding.
-        reopenDecoder();
+IndexRange SoundSourceM4A::readOrSkipSampleFrames(
+        IndexRange frameIndexRange,
+        SampleBuffer::WritableSlice* pOutputBuffer) {
+    auto readableFrames =
+            adjustReadableFrameIndexRangeAndOutputBuffer(
+                    frameIndexRange, pOutputBuffer);
```

Comment:
the same is also true for readOrSkipSampleFrames. Do we need the original IndexRange passed?